### PR TITLE
Tempering edit registration devise form

### PR DESCRIPTION
**Pull request by Giuliamartinelli**
1- Resolved issues about the devise form tempering(method configure_permitted_parameters  was never called).
2-Deleted Views/Users folder which was created by mistake in the beginning. We only kept devise folder!
2-review is dependent destroy from both booking and lesson
3-Resolved possible breaks in the profile ( mentor_lesson, mentor_bookings) and in show(reviews)

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -9,8 +9,10 @@ def profile
     # @reviews = current_user.reviews #non piace a giulia
     @lessons = current_user.lessons # lessons booked as a user
     if current_user.is_mentor
-      @mentor_lesson = current_user.lesson # your lesson as a mentor
-      @mentor_bookings = @mentor_lesson.bookings # mentor bookings of your lesson
+        @mentor_lesson = current_user.lesson # your lesson as a mentor
+        if @mentor_bookings != nil
+        @mentor_bookings = current_user.lesson.bookings # mentor bookings of your lesson
+      end
```
Comment (marcochellogs) [inline app/controllers/pages_controller.rb]: can be refactored

File: app/views/lessons/show.html.erb [hunk 2]
```diff
@@ -27,11 +27,13 @@
   <%= link_to 'Book Lesson', new_lesson_booking_path(@lesson) %>
 <% end %>
 
+<% if @reviews != nil %>
```
Comment (marcochellogs) [inline app/views/lessons/show.html.erb]: refactored?

File: app/views/pages/profile.html.erb [hunk 3]
```diff
@@ -36,9 +39,13 @@
...
         <a href="<%= edit_lesson_path(@mentor_lesson) %>" class="btn btn-primary">Update lesson</a>
+
+
       </div>
     </div>
+    <% end %>
     <%# card mentor bookings  %>
+    <% if @mentor_bookings != nil %>
     <% if @mentor_bookings.count == 0 %>
       <h5>You have 0 bookings</h5>
```
Comment (marcochellogs) [inline app/views/pages/profile.html.erb:49]: to be refactored

File: app/controllers/pages_controller.rb [hunk 4]
```diff
@@ -9,8 +9,10 @@ def profile
     # @reviews = current_user.reviews #non piace a giulia
     @lessons = current_user.lessons # lessons booked as a user
     if current_user.is_mentor
-      @mentor_lesson = current_user.lesson # your lesson as a mentor
-      @mentor_bookings = @mentor_lesson.bookings # mentor bookings of your lesson
+        @mentor_lesson = current_user.lesson # your lesson as a mentor
+        if @mentor_bookings != nil
+        @mentor_bookings = current_user.lesson.bookings # mentor bookings of your lesson
```
Comment (marcochellogs) [inline app/controllers/pages_controller.rb]: can be refactored with ternary opreration. 
example 
@lesson = current_user.lesson
@bookings = @lesson ? @lesson.bookings : current_user.bookings

Comment (marcochellogs) [reply app/controllers/pages_controller.rb]: (disregard the caps)

**Review by marcochellogs — COMMENTED** [review]
All good, some refactoring needed.
Great Job!!
ps. I cannot fall asleep..

File: app/controllers/pages_controller.rb (hunk 4 above)
Comment (Giuliamartinelli) [reply app/controllers/pages_controller.rb]: REFACTORED:
if current_user.is_mentor
      @mentor_lesson = current_user.lesson
      @mentor_bookings = @mentor_lesson ? @mentor_lesson.bookings : current_user.bookings
    end

Comment (Giuliamartinelli) [reply app/controllers/pages_controller.rb]: Tested it on a mentor profile with no bookings and no lessons, it works.
Schermata 2020-10-28 alle 08 10 34

File: app/views/lessons/show.html.erb (hunk 2 above)
Comment (Giuliamartinelli) [reply app/views/lessons/show.html.erb]: REFACTORED:
<% if @reviews.nil? %>
<h5>Missing link to go on new review</h5>
<% else %>
<% @reviews.each do |review| %>
  <h3><%= review.title %></h3>
  <p><%= review.rating %></p>
  <p><%= review.description %></p>
<% end %>
<% end %>

Tested it on a new lesson with no reviews.
Schermata 2020-10-28 alle 08 15 06

File: app/views/pages/profile.html.erb (hunk 3 above)
Comment (Giuliamartinelli) [reply app/views/pages/profile.html.erb:49]: REFACTORED:
<% if @mentor_bookings.nil? %>
      <h5>You have 0 bookings</h5>
    <% else %>
      <h5>You have <%= @mentor_bookings.count %> bookings:</h5> <%# we need always more than 1 booking as seed %>
      <% @mentor_bookings.each do |booking| %>
        <%= booking.date %>
        <%= booking.user.name %> <%# student's name  %>
    <% end %>
  <% end %>

Tested it on empty bookings profile
Schermata 2020-10-28 alle 08 18 25

File: app/controllers/pages_controller.rb [hunk 5]
```diff
@@ -9,12 +9,11 @@ def profile
     # @reviews = current_user.reviews #non piace a giulia
     @lessons = current_user.lessons # lessons booked as a user
     if current_user.is_mentor
-        @mentor_lesson = current_user.lesson # your lesson as a mentor
-        if @mentor_bookings != nil
-        @mentor_bookings = current_user.lesson.bookings # mentor bookings of your lesson
-      end
+      @mentor_lesson = current_user.lesson
+      @mentor_bookings = @mentor_lesson ? @mentor_lesson.bookings : current_user.bookings
```
Comment (marcochellogs) [inline app/controllers/pages_controller.rb:13]: this is user bookings regardless of mentor or student

Comment (marcochellogs) [reply app/controllers/pages_controller.rb:13]: take out the first if

Comment (marcochellogs) [reply app/controllers/pages_controller.rb:13]: ok maybe we need to review this together, I might be missing something

**Review by marcochellogs — COMMENTED** [review]
Let's review it together

**Pull request merged**